src/OrchardCore/OrchardCore.Media.Core/DefaultMediaFileStore.cs

```diff
@@ -178,7 +179,30 @@ public virtual async Task<string> CreateFileFromStreamAsync(string path, Stream
 
         public virtual string MapPathToPublicUrl(string path)
         {
-            return _cdnBaseUrl + _requestBasePath + "/" + _fileStore.NormalizePath(path);
+
+            if (Uri.TryCreate(path, UriKind.Absolute, out _))
```

Comment:
This will fail on linux and consider most paths as absolute, due to what linux considers an absolute path.

https://github.com/dotnet/runtime/issues/22718